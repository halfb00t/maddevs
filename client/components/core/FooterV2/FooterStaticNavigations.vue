<template>
  <div class="footer-subnavbar">
    <p class="footer-subnavbar__company-name">
      Â© Mad Devs - {{ currentYear }}
    </p>
    <div class="footer-subnavbar__nav-list">
      <NuxtLink
        v-for="item in staticsPagesLinks"
        :key="item.link"
        :to="item.link"
        class="footer-subnavbar__nav-list-item"
        target="_blank"
      >
        {{ item.title }}
      </NuxtLink>
    </div>
  </div>
</template>

<script>
import { staticsPagesLinks } from '@/data/footerNavigation'

export default {
  name: 'FooterStaticNavigations',
  data() {
    return {
      staticsPagesLinks,
      currentYear: new Date().getFullYear(),
    }
  },
}
</script>

<style lang="scss" scoped>
.footer-subnavbar {

  @media screen and (max-width: 992px) {
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
    margin-top: 30px;
  }

  &__company-name {
    opacity: 0.7;
    font-size: 12px;
    color: #96969C;
    @media screen and (min-width: 992px) {
      margin-top: 105px;
      @include font('inter', 14px, 400);
    }
  }

  &__nav {
    &-list {
      @media screen and (min-width: 992px) {
        margin-top: 30px;
      }
    }

    &-list-item {
      color: #96969C;
      display: block;
      @include font('inter', 14px, 400);
      opacity: 0.7;

      @media screen and (max-width: 1320px) {
        font-size: 12px;
      }
    }
  }
}
</style>
