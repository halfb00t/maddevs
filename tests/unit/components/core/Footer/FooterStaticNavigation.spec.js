import { render } from '@testing-library/vue'
import FooterStaticNavigations from '@/components/core/Footer/FooterStaticNavigations'

const directives = { 'lazy-load': jest.fn() }
const stubs = ['NuxtLink']
describe('FooterNavbarColumn', () => {
  it('Should correct render footer static navigation in footer', () => {
    const { container } = render(FooterStaticNavigations, { directives, stubs })
    expect(container)
      .toMatchSnapshot()
  })
})
