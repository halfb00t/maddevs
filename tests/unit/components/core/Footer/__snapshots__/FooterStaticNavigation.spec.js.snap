// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterNavbarColumn Should correct render footer static navigation in footer 1`] = `
<div>
  <div
    class="footer-subnavbar"
  >
    <div
      class="footer-subnavbar__nav-list footer-subnavbar__nav-list--pdf"
    >
      <a
        class="footer-subnavbar__nav-list-item footer-subnavbar__nav-list-item--pdf"
        href="https://maddevsio.s3.eu-west-1.amazonaws.com/pdf/about-mad-devs-upd.pdf"
        target="_blank"
      >
        
      About Us.pdf
    
      </a>
      <a
        class="footer-subnavbar__nav-list-item footer-subnavbar__nav-list-item--pdf"
        href="https://maddevsio.s3.eu-west-1.amazonaws.com/pdf/mad-devs-work-process.pdf"
        target="_blank"
      >
        
      Work process.pdf
    
      </a>
      <a
        class="footer-subnavbar__nav-list-item footer-subnavbar__nav-list-item--pdf"
        href="https://maddevsio.s3.eu-west-1.amazonaws.com/pdf/mad-devs-cooperation-models.pdf"
        target="_blank"
      >
        
      Cooperation models.pdf
    
      </a>
      <a
        class="footer-subnavbar__nav-list-item footer-subnavbar__nav-list-item--pdf"
        href="https://maddevsio.s3.eu-west-1.amazonaws.com/pdf/mad-devs-brand-book-upd.pdf"
        target="_blank"
      >
        
      Brand Book.pdf
    
      </a>
    </div>
     
    <div
      class="footer-subnavbar-container"
    >
      <div
        class="footer-subnavbar__nav-list footer-subnavbar__nav-list--pages"
      >
        <nuxtlink-stub
          class="footer-subnavbar__nav-list-item footer-subnavbar__nav-list-item--pages"
          target="_blank"
          to="/gdpr/"
        >
          
        GDPR Compliance Commitment
      
        </nuxtlink-stub>
        <nuxtlink-stub
          class="footer-subnavbar__nav-list-item footer-subnavbar__nav-list-item--pages"
          target="_blank"
          to="/nda/"
        >
          
        Non-Disclosure Agreement (NDA)
      
        </nuxtlink-stub>
      </div>
       
      <p
        class="footer-subnavbar__company-name"
      >
        
      Â© Mad Devs - 2022
    
      </p>
    </div>
  </div>
</div>
`;
